skrub/_gap_encoder.py

```diff
@@ -389,8 +394,12 @@ def get_feature_names_out(
         for i in range(n_components):
             x = encoding[:, i]
             labels = vocabulary[np.argsort(-x)[:n_labels]]
-            topic_labels.append(labels)
-        topic_labels = [prefix + ", ".join(label) for label in topic_labels]
+            label = ", ".join(labels)
+            # Avoid having twice the same name for the different features
+            if label in topic_labels:
+                label += ", {:}".format(i)
```

Comment:
just out of curiosity why not `f", {i}"`?

also we might go for a different separator to make it more obvious this isn't one of the top words in the topic but something that was added to avoid duplicate columns eg ` (i)` instead of `, i`. this is really an edge case so it doesn't really matter though